src/utils/utils.ts

```diff
@@ -13,16 +16,63 @@ const snakeCase = (word: string): string => {
     .join("_");
 };
 
-const buildQueryString = (params: Params): string => {
+const snakeCaseObjectKeys = (obj: any) => {
+  let key;
+  const keys = Object.keys(obj);
+  let n = keys.length;
+  const newObj = {};
+  while (n--) {
+    key = keys[n];
+    newObj[snakeCase(key)] = obj[key];
+  }
+
+  return newObj;
+};
+
+const tokenizeData = async (patientInfo, fullscriptOptions) => {
+  try {
+    const { env, domain } = fullscriptOptions;
+    const fsDomain = domain ?? FULLSCRIPT_DOMAINS[env];
+
+    const tokenizedInfo = await fetch(`${fsDomain}/api/embeddable/tokenize`, {
+      method: "POST",
+      body: JSON.stringify({ data: patientInfo }),
+      headers: { "Content-Type": "application/json" },
+    }).then(res => res.json());
+
+    if (typeof tokenizedInfo.data_token !== "string") throw new Error("Invalid response");
+
+    return tokenizedInfo.data_token;
+  } catch (error) {
+    return null;
+  }
+};
+
+const buildQueryString = async (params: Params): Promise<string> => {
   if (!Object.keys(params) || Object.keys(params).length === 0) return "";
-  return Object.keys(params).reduce((queryString, key) => {
-    let newParam = `${snakeCase(key)}=${params[key]}`;
 
-    if (typeof params[key] !== "string") {
-      newParam = Object.keys(params[key]).reduce(
+  let { patient: patientInfo, fullscriptOptions, ...exposedParams } = params;
+  let startingQueryString = "?";
+
+  if (patientInfo) {
+    const snakeCasePatient = snakeCaseObjectKeys(patientInfo);
+    const dataToken = await tokenizeData(snakeCasePatient, fullscriptOptions);
+
+    if (!dataToken) {
+      exposedParams = { patient: snakeCasePatient, ...exposedParams };
```

Comment:
in the off-chance that the tokenization function fails, we 're-expose' the patient info

Reply:
do we want to return an error though? The query string will also contain 'at least' the secret token since it's required.

Reply:
Good point! 

I'm worried that exposing the personal info again when `tokenizeData` fails could be a loophole for a security issue. Meaning our library is only 99.99% secured. And since this an open source, integration partners might be able to see this minor thing.

@SeanKG , any thoughts?

Reply:
Updated this based on our discussion to pass a `null` data token instead of re-exposing the patient data